### PR TITLE
Create cisco_xr_show_run_hostname

### PR DESCRIPTION
**Pull request by jonathondamidco**
This template is required due to the inconsistency of availability of the commands `show version`, `show version brief` and `show version detail` or the hostnames therein with different XR versions. `show run hostname` is pretty straight-forward and should always return just the hostname.

### CODE REVIEW

**Comment by jonathondamidco** [comment]
You might also be asking why a textfsm template is even needed for this, since it's such a trivial thing to grab out of the CLI output. The answer is Nautobot Device Onboarding 4.0 needs textfsm to parse it out, since you can't display Cisco configs as JSON as you can with platforms like JunOS.

**Comment by jonathondamidco** [comment]
@jmcgill298 Neither of those commands are available. Without it being in the output of `show version`, I've been looking for any other show command that displays it.

File: ntc_templates/templates/cisco_xr_show_run_hostname.textfsm [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Value HOSTNAME (\S+)
+
+Start
+  # Capture date output by XR by 'show run hostname' first
+  ^\S+ \S+ \S+ .*$$
+  # Capture hostname
+  ^hostname ${HOSTNAME}$$
+  ^\s*$$
+  ^. -> Error
```
Comment (jmcgill298) [inline ntc_templates/templates/cisco_xr_show_run_hostname.textfsm:9]:
```suggestion
  ^. -> Error
  
```

File: tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.raw [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Thu Aug 29 16:30:30.674 CDT
+hostname rtr-01
```
Comment (jmcgill298) [inline tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.raw:2]:
```suggestion
hostname rtr-01

```

File: tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+parsed_sample:
+  - hostname: "rtr-01"
```
Comment (jmcgill298) [inline tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.yml:3]:
```suggestion
  - hostname: "rtr-01"
  
```

File: ntc_templates/templates/cisco_xr_show_run_hostname.textfsm [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
```
Comment (jmcgill298) [inline ntc_templates/templates/cisco_xr_show_run_hostname.textfsm:10]:
```suggestion

```

File: tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+
```
Comment (jmcgill298) [inline tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.yml:4]:
```suggestion

```

File: ntc_templates/templates/cisco_xr_show_run_hostname.textfsm (hunk 1 above)
Comment (jmcgill298) [inline ntc_templates/templates/cisco_xr_show_run_hostname.textfsm:9]:
```suggestion
  ^. -> Error

```

File: tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.yml (hunk 3 above)
Comment (jmcgill298) [inline tests/cisco_xr/show_run_hostname/cisco_xr_show_run_hostname.yml:3]:
```suggestion
  - hostname: "rtr-01"

```

**Review by jmcgill298 — APPROVED** [review]

**Comment by jmcgill298** [comment]
thanks @jonathondamidco

**Pull request merged**